### ISSUE TITLE
Feedback: why and how are you using this lib?

### ISSUE THREAD

**Issue by willdurand** [willdurand/EmailReplyParser#15]
Hi!

I must admit that I never used this library myself, I just took a couple hours to port the GitHub's email_reply_parser library because it looked cool. Earlier this week, I decided to read the code again, and it was quite awful, so I refactored the lib, and I shipped [2.0.0](https://github.com/willdurand/EmailReplyParser/releases/tag/2.0.0). I hope you will like it.

Anyway, if **you are a user**, I'd love to hear from you! **Could you please write a comment below**, telling me more about your **use cases**, **why** and **how** you are using this lib, and maybe some **ideas** or things you would love to see in it?

:heart: Thank you! :heart: 

Will

**Comment by pbrzoski** [comment]
Is exist any parser for email forward ?

**Comment by cristobal-wetzig** [comment]
You rock!


We use this in combination with postmarkapp service to extract the signatures from incomming mail.

I noticed you have french support in the regexp. How about swedish? Or some way to add them.

example from apple mail client:
5 mar 2015 kl. 12:03 skrev {joe}

**Comment by pbrzoski** [comment]
If you want, you can create pull request with swedish. 

Regexps for other language, are in file
src/EmailReplyParser/Parser/EmailParser.php

**Comment by willdurand** [comment]
Yes, adding swedish is ok. But there is also a method for that IIRC.

**Comment by willdurand** [comment]
Thanks for the kind words btw :)

**Comment by cristobal-wetzig** [comment]
@willdurand m8 if you ever in Stockholm, a cold beer is waiting for you.

**Comment by rmccue** [comment]
I've been using EmailReplyParser for quite a while now, but must have missed this issue previously!

We're using it in a WordPress plugin called [Falcon](https://github.com/rmccue/Falcon) that allows people to reply by email. Super handy when using a WordPress site as an internal company message board (with something like [P2](http://p2theme.com/)).

EmailReplyParser means we can have minimal effort when replying for a basically seamless integration. Thanks for maintaining this. :heart:

**Comment by miladr** [comment]
I'm going to use it for our ticketing service.
As you may know major services like Zendesk use a string like "## Please type your reply above this line ##" and when user click reply button this string is above all quoted lines.
It's a good idea to accept a string and put all below lines in a quoted fragment.

**Comment by lflaszlo** [comment]
Hi, thanks for your work on this lib.

We use this lib for our internal ticketing system, we parse emails inputs directly to tickets and we "clean" these emails thanks to EmayReplayParser.

Regards,

**Comment by perajovic** [comment]
We use it as a very important component of our SaaS business -- https://supportyard.com. And we love it!

**Comment by willdurand** [comment]
Thank you for your replies.

**Comment by sigginet** [comment]
I use it in our production code to parse e-mail messages.
It's immensely valuable, but I may need to add more detection scenarios for more languages since we deal with a lot of different languages.

**Comment by dbr0** [comment]
We use it in our ticketing system to parse the replies we get through emails.

**Comment by blyleven** [comment]
Thank you so much for this! Amsterdam is very happy with your work. Using this to send chatbox messages via email.

**Comment by willdurand** [comment]
Dear users, [2.6.0](https://github.com/willdurand/EmailReplyParser/releases/tag/2.6.0) is out today. Thanks!

**Comment by thebordella** [comment]
This class looks great! Question though - we need something like this to parse e-mail replies inbound from Sendgrid. We want to extract just the reply and not all the quoted content in the chain.

But this class only parses plaintext email, right? When users reply they are using their own mail clients. I know that Sendgrid provides the plaintext content if available, and in my tests I see that Gmail and Thunderbird do by default include the plaintext alternative. But I don't know whether to assume most email clients will. 

What do people do about this? How do we best ensure we can feed plaintext replies into this parser? Thank for any advice!

**Comment by spidgorny** [comment]
@thebordella 
```
$plain = html_entity_decode(
    trim(
        strip_tags($html)
    )
);
```

**Comment by SL-Gundam** [comment]
Hi

I'm a developer for the [EmailReporting](https://github.com/mantisbt-plugins/EmailReporting) project which is a plugin for [MantisBT](https://github.com/mantisbt/mantisbt).

I'm planning to use this library to process incoming emails more effectively and create better tickets or notes for MantisBT

I'm already using [Markdownify ](https://github.com/Elephant418/Markdownify) to process html emails to markdown. I wonder how well EmailReplyParser handles that content. We'll see. If succesfull this will retire more simple existing code in EmailReporting

**Comment by harikt** [comment]
Hi @willdurand ,
I am not using it myself, but was trying to help someone as other notified similar to github notification reply to email sort of stuff.

1. Probably finding attachments would be a nice feature.
2. As already mentioned attaching more parsers would be a nice feature to have.

**Comment by billynoah** [comment]
Using this to import emails into a ticket and rma system.  Had written something close myself until I realized how many various formats are in play.  A quick search brought me here from [this stackoverflow answer](https://stackoverflow.com/questions/14916618/extract-email-message-itself-from-all-its-prior-messages-and-meta-data-sendgrid/14916833)

Your work is much appreciated!

**Comment by barryvdh** [comment]
I'm using this to process inbound emails for tickets/replies, so useful to just get the reply part.
I added this lib to https://github.com/beyondcode/laravel-mailbox/pull/7 to make it easier to process Inbound replies in Laravel.

What would make it easier to maintain perhaps, is making it possible to add custom signatures manually (eg `$parser->addHeaderRegex($pattern)`).

Are you still maintaining this, because it still looks like one of the best available packages in PHP.

**Comment by barryvdh** [comment]
Oh my bad, there is actually `getQuoteHeadersRegex()` and `gsetQuoteHeadersRegex()` already :)

**Comment by barryvdh** [comment]
And just as a FYI, I did some digging and most libraries support only English (like Github). The discourse reply trimmer is more extensive (and multi language) it seems: https://github.com/discourse/email_reply_trimmer
So perhaps for a future version, we could take a look at which regexes etc they use, or copy over the tests.

**Comment by willdurand** [comment]
> Are you still maintaining this, because it still looks like one of the best available packages in PHP.

I am, at a very very low pace.

Thanks for using this lib!

**Comment by glensc** [comment]
We use this in [Eventum Issue Tracker][eventum] to collapse quoted replies like GMail does:

- https://github.com/eventum/eventum/pull/143

[eventum]: https://github.com/eventum/eventum

**Comment by philippejadin** [comment]
I'm using it with great success for the open source groupware for collectives called Agorakit : https://github.com/agorakit/agorakit

And, really thank you for this tool, in short, the best option for php !

**Comment by voicecode-bv** [comment]
Still using this package, thank you so much for it. We're using it to allow users to send emails that will be added to a chatbox.
